boot/bootutil/src/bootutil_misc.c

```diff
@@ -144,11 +146,11 @@ boot_magic_code(const uint32_t *magic)
 {
     int i;
 
-    if (memcmp(magic, boot_img_magic, sizeof boot_img_magic) == 0) {
+    if (memcmp(magic, boot_img_magic, BOOT_MAGIC_SZ) == 0) {
         return BOOT_MAGIC_GOOD;
     }
 
-    for (i = 0; i < 4; i++) {
+    for (i = 0; i < BOOT_MAGIC_SZ; i++) {
         if (magic[i] == 0xffffffff) {
```

Comment:
It is interesting that we only compare against the good magic or 0xffffffff. I wonder, in general, about the handling of partial writes to flash. Generally, when a write is interrupted by power loss, the resulting data will contain a combination of:
- The old data, in this case 1 bits.
- The new data, zero bits.
and the combination that is read can be different, even on subsequent reads, but even worse, can change as temperature and voltage change.